sdk/nodejs/cmd/pulumi-language-nodejs/main_test.go

```diff
@@ -167,3 +168,56 @@ func TestGetRequiredPlugins(t *testing.T) {
 		"bar": "v4.5.6",
 	}, actual)
 }
+
+func TestGetRequiredPluginsSymlinkCycles(t *testing.T) {
+	t.Parallel()
+
+	dir := t.TempDir()
+
+	files := []struct {
+		path    string
+		content string
+	}{
+		{
+			filepath.Join(dir, "node_modules", "@pulumi", "foo", "package.json"),
+			`{ "name": "@pulumi/foo", "version": "1.2.3", "pulumi": { "resource": true } }`,
+		},
+		{
+			filepath.Join(dir, "node_modules", "@pulumi", "bar", "package.json"),
+			`{ "name": "@pulumi/bar", "version": "4.5.6", "pulumi": { "resource": true } }`,
+		},
+		{
+			filepath.Join(dir, "node_modules", "@pulumi", "baz", "package.json"),
+			`{ "name": "@pulumi/baz", "version": "4.5.6", "pulumi": { "resource": false } }`,
+		},
+		{
+			filepath.Join(dir, "node_modules", "malformed", "tests", "malformed_test", "package.json"),
+			`{`,
+		},
+	}
+	for _, file := range files {
+		err := os.MkdirAll(filepath.Dir(file.path), 0755)
+		assert.NoError(t, err)
+		err = os.WriteFile(file.path, []byte(file.content), 0600)
+		assert.NoError(t, err)
```

Comment:
This is all part of the setup.
We should probably use `require` here
because if any of this fails, there's no point running the rest of the test.

```suggestion
		err := os.MkdirAll(filepath.Dir(file.path), 0755)
		require.NoError(t, err)
		err = os.WriteFile(file.path, []byte(file.content), 0600)
		require.NoError(t, err)
```

Reply:
Yeh good point, this was copied from the test above.